### PR TITLE
Feature/and or querysetrules

### PR DESCRIPTION
**Pull request by shwetabhandare**
## Description
Add and/or query set rules to Django drip.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### CODE REVIEW

File: drip/drips.py [hunk 1]
```diff
@@ -195,6 +195,21 @@ def walk(self, into_past: int = 0, into_future: int = 0):
         return walked_range
 
     def apply_queryset_rules(self, qs: str) -> str:
+        return (self.apply_and_queryset_rules(qs) | self.apply_or_queryset_rules(qs))
```
Comment (kaozdl) [inline drip/drips.py:198]:
```suggestion
        return (
            self.apply_and_queryset_rules(qs) |
            self.apply_or_queryset_rules(qs)
        )
```

**Review by kaozdl — DISMISSED** [review]

**Review by jiss2891 — COMMENTED** [review]
This is looking good, does the test pass ok?

File: drip/drips.py [hunk 2]
```diff
@@ -195,6 +195,24 @@ def walk(self, into_past: int = 0, into_future: int = 0):
...
+
+    def apply_or_queryset_rules(self, qs: str) -> str:
+        query = None
+        rule_set = self.drip_model.queryset_rules.filter(rule_type='or')
+        for rule in rule_set:
+            kwargs = rule.filter_kwargs(qs, now=self.now)
+            query_or = Q(**kwargs)
+            if query is None:
+                query = qs.none()
+            query = query | query_or
```
Comment (jiss2891) [inline drip/drips.py:211]: I think this can be done with:

```python
query |= query_or
```

File: drip/drips.py [hunk 3]
```diff
@@ -195,6 +195,24 @@ def walk(self, into_past: int = 0, into_future: int = 0):
...
+        qs = qs.filter(query) if query is not None else qs.none()
```
Comment (jiss2891) [inline drip/drips.py:212]: This line could be:

```python
qs = qs.filter(query) if query else qs.none()
```

Comment (kaozdl) [reply drip/drips.py:212]: I think in this case having the `is None` actually adds clarity to what query can be.

Comment (jiss2891) [reply drip/drips.py:212]: Sounds good to me.

**Review by kaozdl — APPROVED** [review]

**Review by brunomichetti — CHANGES_REQUESTED** [review]
Hello, thanks for contributing!
Can you modify the version of the package in `drip/__init__.py`?
Since it's a non-breaking change, you should bump the second digit as we have defined in CONTRIBUTING.md

Thanks again

**Comment by shwetabhandare** [comment]
@kaozdl - closing this PR and making a new one with the changes to make sure OR rules work when there are no AND rules.

**Pull request closed without merging**